common/flatpak-context.c

```diff
@@ -852,31 +916,6 @@ flatpak_context_parse_filesystem (const char             *filesystem_and_mode,
   return FALSE;
 }
 
-/* Note: This only works with valid keys, i.e. they passed flatpak_context_parse_filesystem */
-static gboolean
-flatpak_filesystem_key_in_home (const char *filesystem)
-{
-  /* "home" is definitely in home */
-  if (strcmp (filesystem, "home") == 0)
-    return TRUE;
-
-  /* All the other special fs:es are non-home.
-   * Note: This considers absolute paths that are in the homedir as non-home.
-   */
-  if (g_strv_contains (flatpak_context_special_filesystems, filesystem) ||
-      g_str_has_prefix (filesystem, "/"))
-    return FALSE;
-
-  /* Files in xdg-run are not in home */
-  if (g_str_has_prefix (filesystem, "xdg-run"))
-    return FALSE;
-
-  /* All remaining keys (~/, xdg-data, etc) are considered in home,
-   * Note: technically $XDG_HOME_DATA could point outside the homedir, but we ignore that.
-   */
-  return TRUE;
-}
-
 static void
 flatpak_context_take_filesystem (FlatpakContext        *context,
                                  char                  *fs,
```

Comment:
Should `host-reset` imply `host => NONE`?

So for example if we have

```
flatpak run --filesystem=host --nofilesystem=host:reset ...
flatpak run --filesystem=host --nofilesystem=host:reset --nofilesystem=host ...
```

are those equivalent (with the host filesystem ending up unshared), or does the first one reset inherited filesystems *and* unshare host filesystems?

If we are using `--nofilesystem=host:reset` in flatpak-builder as a way to make sure it gracefully degrades to the equivalent of `--nofilesystem=host` (with a warning) in older Flatpak releases, then I think least-astonishment is for `--nofilesystem=host:reset` to be exactly equivalent to `--nofilesystem=host:reset --nofilesystem=host` or `--nofilesystem=host --nofilesystem=host:reset`.

If someone really, really wants the semantics of "stop inheriting filesystems, but then share `host`", they can still have that, via `--nofilesystem=host:reset --filesystem=host`.

Reply:
I think `host-reset` should imply `host => NONE`, and in my latest push, it does.